### ISSUE TITLE
FFaker::AddressAU.country_code returns random results

### ISSUE THREAD

**Issue by rabinshr** [ffaker/ffaker#453]
Trying to use ffaker module and noticed the `country_code` returns random results. See:
![image](https://user-images.githubusercontent.com/1313863/89771945-85f66d00-db44-11ea-9b7c-ca2c6440e31b.png)

I would expect `FFaker::AddressAU.country_code` to return `AU` and `FFaker::AddressUS.country_code` to return `US`

**Comment by Volosh1n** [comment]
Hi, @rabinshr!
Do you mean that AddressUS should always return 'US'?

**Comment by rabinshr** [comment]
Yes, that was my expectation. 
What is the expected result here?

**Comment by Volosh1n** [comment]
Actually, all those `Address...` modules returns names in specified language. I mean, AddressUS returns names in English, while, for example, AddressCH returns address names in Chineese.
`country_code` returns values from almost the same collection of country codes accross whole gem.
If some methods returns only one value every time, then this FFaker's method becomes meaningless.

**Comment by rabinshr** [comment]
@Volosh1n 
right, so `AddressUS` `AddressAU` etc refer to language rather than country.
But then, the expectation mismatches when i do `FFaker::AddressAU.state_abbr` which only returns australian state abbreviations 
```
2.6.3 :027 > FFaker::AddressAU.state_abbr
 => "SA" 
2.6.3 :028 > FFaker::AddressAU.state_abbr
 => "ACT" 
2.6.3 :029 > FFaker::AddressAU.state_abbr
 => "VIC" 
2.6.3 :030 > FFaker::AddressAU.state_abbr
 => "TAS" 
2.6.3 :031 > FFaker::AddressAU.state_abbr
 => "ACT" 
```
So in this case, it actually looks like `FFaker::AddressAU` is referring to australian address 🤔

**Comment by Volosh1n** [comment]
I think it is because `AddressAU` includes `FFaker::Address` and does not implement it's own `country_code` method (so, uses `FFaker::Address`'s `country_code` method). Idk, maybe we can keep the same logic. Not a big deal, but it's about generation some random values accros many tries, not only 'AU'.
Maybe @marocchino think otherwise. Anyway, thank you for this remark.

**Comment by rabinshr** [comment]
No worries.
We have had quite a bit of confusion in terms of expectation.
Will wait to see what others think.
Thanks for the replies 🙇

**Comment by hmaack** [comment]
When I get into the user perspective, I completely agree to @rabinshr opinion. It's OOP inheritance.

From my perspective: 

1. ```FFaker::Address.country_code``` is perfect for random country codes
2. Any specific `Address` Module should return an exact country code like `FFaker::AddressAU.country_code` => `AU` - https://countrycode.org/australia or should not have this as a public API method

I think we can make the API clear, predictable if we update the `REFERENCE.md` and kick the country_code for the specific country modules or we update all specific modules to return a fix country_code. Any opinion about that?

**Comment by shafy** [comment]
Just to add a +1: I was also expecting the same thing as @rabinshr. In fact, I was expeting `AddressCH` to return Swiss addresses, and `AddressCHDE` to return addresses in the German-speaking part of Switzerland, `AddressCHFR` in the Swiss speaking part etc. Oddly, the cantons (`.canton_abbr`) are returned as expected, but the rest isn't.

**Comment by maysam** [comment]
if you want `FFaker::AddressAU.country_code` to return `AU` all the time, you should just hardcode `AU`.

`FFaker::AddressAU.country_code` is supposed to give you a random `country_code` which is valid in `australian` context

**Comment by rabinshr** [comment]
@maysam , thanks for your comment.

>FFaker::AddressAU.country_code is supposed to give you a random country_code which is valid in australian context

which I believe would only be `AU` for australian context. 

![image](https://user-images.githubusercontent.com/1313863/148296080-f000055a-e20d-4940-af84-bce05dfdb022.png)
^ Help me understand how the above country codes are valid in the context of australia/US

**Comment by marocchino** [comment]
It is a valid point, considering that countries like China have multiple country codes; however, since changing the behavior would cause regression for users who expected the existing behavior, I believe we need to approach this with caution.
